pravega-cli/src/main/java/io/pravega/tools/pravegacli/commands/admin/PasswordFileCreatorCommand.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.tools.pravegacli.commands.admin;
+
+import com.google.common.base.Strings;
+import io.pravega.tools.pravegacli.commands.CommandArgs;
+import lombok.Cleanup;
+import io.pravega.tools.pravegacli.commands.Command;
+import io.pravega.controller.server.rpc.auth.StrongPasswordProcessor;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.lang.ArrayIndexOutOfBoundsException;
+import java.util.List;
+import java.util.ArrayList;
+
+public class PasswordFileCreatorCommand extends Command {
+    static final String COMPONENT = "admin";
+    public PasswordFileCreatorCommand(CommandArgs args){super(args);}
+    private String toWrite;
+
+    public String getToWrite(){
+        return this.toWrite;
+    }
+
+    @Override
+    public void execute() throws InvalidKeySpecException, NoSuchAlgorithmException, IOException {
+        try {
+            ensureArgCount(2);
+            String targetFileName = getTargetFilename(getCommandArgs().getArgs());
+            String userDetails = getUserDetails(getCommandArgs().getArgs());
+            createPassword(userDetails);
+            WriteToFile(targetFileName, toWrite);
+        }
+        catch (Exception e){
+            System.err.println(e.getMessage());
+        }
+    }
+
+    public String getTargetFilename(List<String> userInput) {
+        return userInput.get(0);
+    }
+
+    public String getUserDetails(List<String> userInput){
+        String userDetails = userInput.get(1);
+        if((userDetails.split(":")).length==3){
+            return userDetails;
+        }
+        else throw new IllegalArgumentException("The user detail entered is not of the format uname:pwd:acl");
+    }
+
+    public void createPassword(String userDetails) throws NoSuchAlgorithmException, InvalidKeySpecException {
+        String[] lists = parseUserDetails(userDetails);
+        toWrite = generatePassword(lists);
+    }
+
+    private String[] parseUserDetails(String userDetails){
+        return userDetails.split(":");
+    }
+
+    private String generatePassword(String[] lists) throws NoSuchAlgorithmException, InvalidKeySpecException{
+        StrongPasswordProcessor passwordEncryptor = StrongPasswordProcessor.builder().build();
+        return lists[0] + ":" + passwordEncryptor.encryptPassword(lists[1]) + ":" + lists[2] + ";";
+    }
+
+    private void WriteToFile (String targetFileName, String toWrite) throws IOException {
```

Comment:
In Java, method names start by lower case by convention.

Reply:
Same here. Resolve the conversation when the comment is addressed, please.

Reply:
Okay. I have made the change but not pushed the commit yet. Will keep this in mind.